controllers/scopeinstance_controller.go

```diff
@@ -78,11 +79,37 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	log.Log.Info("Reconciling ScopeInstance", "namespaceName", req.NamespacedName)
 
-	in := &operatorsv1.ScopeInstance{}
-	if err := r.Client.Get(ctx, req.NamespacedName, in); err != nil {
-		return ctrl.Result{}, err
+	existingIn := &operatorsv1.ScopeInstance{}
+	if err := r.Client.Get(ctx, req.NamespacedName, existingIn); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Perform reconciliation
+	reconciledIn := existingIn.DeepCopy()
+	res, reconcileErr := r.reconcile(ctx, reconciledIn)
+
+	// Update the status subresource before updating the main object. This is
+	// necessary because, in many cases, the main object update will remove the
+	// finalizer, which will cause the core Kubernetes deletion logic to
+	// complete. Therefore, we need to make the status update prior to the main
+	// object update to ensure that the status update can be processed before
+	// a potential deletion.
```

Comment:
I don't *think* so. This was pretty much pulled straight from rukpak and updated to work with our controllers. My understanding is that we are applying any status updates before we update the main object to remove any finalizers during a delete request. I believe this order prevents a race condition if the updates were done in a reverse order.

The place where this happens in rukpak is here: https://github.com/operator-framework/rukpak/blob/214a38947a0f3994f523c65b7804c759bbd18855/internal/provisioner/plain/controllers/bundle_controller.go#L82-L102

If we feel it is best, I can also remove this comment but I thought it might be nice to have the comment to explain what is happening.

Reply:
Are we using finalizers?

Reply:
As of now we are not